File: 2-second-year-2022-2023/Programming-languages-and-laboratory/Java/Exercises/21_compito_java_2018_04_06b/GestioneTorneo.java
import java.util.*;
import java.io.*;

public class GestioneTorneo {
	static List<Giocatore> giocatori = new LinkedList<Giocatore>();

	public static void main(String[] args) {

		// Punto 1, memorizzazione giocatori

		try {

			BufferedReader br = new BufferedReader(new FileReader("giocatori.txt"));
			String line = br.readLine();

			while (line != null) {
				int codiceGiocatore = Integer.parseInt(line);
				String tipo = br.readLine();
				String nomeCognomeGiocatore = br.readLine();

				line = br.readLine();

				if (tipo.equals("testa-di-serie")) {
					StringTokenizer tok = new StringTokenizer(line);
					int ranking = Integer.parseInt(tok.nextToken());
					int eta = Integer.parseInt(tok.nextToken());
					boolean professionista = Boolean.parseBoolean(tok.nextToken());

					Giocatore g = new testaDiSerie(codiceGiocatore, tipo, nomeCognomeGiocatore, ranking, eta,
							professionista);
					giocatori.add(g);

					line = br.readLine();

				} else {
					String nomeCircolo = line;
					line = br.readLine();
					StringTokenizer tok = new StringTokenizer(line);
					int numeroSetVinti = Integer.parseInt(tok.nextToken());
					int numeroPartiteDisputate = Integer.parseInt(tok.nextToken());

					Giocatore g = new Normale(codiceGiocatore, tipo, nomeCognomeGiocatore, nomeCircolo, numeroSetVinti,
							numeroPartiteDisputate);
					giocatori.add(g);

					line = br.readLine();

				}

				line = br.readLine();
			}

			br.close();

		} catch (IOException e) {
			System.err.println(e);
		} catch (Exception e) {
			System.err.println(e);
		}

		// Punto 2, memorizzazione partite

		try {

			BufferedReader br = new BufferedReader(new FileReader("partite.txt"));
			String line = br.readLine();

			while (line != null) {
				StringTokenizer tok = new StringTokenizer(line);
				int codicePrimoGiocatore = Integer.parseInt(tok.nextToken());
				int codiceSecondoGiocatore = Integer.parseInt(tok.nextToken());
				line = br.readLine();
				tok = new StringTokenizer(line);
				int setVintiPrimoGiocatore = Integer.parseInt(tok.nextToken());
				int setVintiSecondoGiocatore = Integer.parseInt(tok.nextToken());

				Partita p = new Partita(codicePrimoGiocatore, codiceSecondoGiocatore, setVintiPrimoGiocatore,
						setVintiSecondoGiocatore);

				for (Giocatore g : giocatori) {
					if (p.codicePrimoGiocatore == g.codiceGiocatore) {
						g.partite.add(p);
					}

					if (p.codiceSecondoGiocatore == g.codiceGiocatore) {
						Partita par = new Partita(codiceSecondoGiocatore, codicePrimoGiocatore,
								setVintiSecondoGiocatore, setVintiPrimoGiocatore);
						g.partite.add(par);
					}

				}

				line = br.readLine();

			}

			br.close();

		} catch (

		IOException e) {
			System.err.println(e);
		} catch (Exception e) {
			System.err.println(e);
		}

		// Punto 3, stampa elenco giocatori

		System.out.println("nome\tcodice\ttipo\tranking\tetà\tprofessionista\tcircolo\tn. set vinti\tn. partite");

		for (Giocatore g : giocatori) {
			System.out.println(g);
		}

		System.out.println("");

		// Punto 4, stampa nome ed elenco partite giocatore

		for (Giocatore gio : giocatori) {
			System.out.println(gio.nomeCognome + " " + gio.partite);
		}

		System.out.println("");

	}
}

File: II - Linguaggi di programmazione e laboratorio/Esame/Python/21b_compito_java_20180406/GestioneTorneo.py
from Giocatore import Giocatore
from TestaDiSerie import TestaDiSerie
from Normale import Normale
from Incontro import Incontro


class GestioneTorneo:
    def main():

        giocatori = []
        cod_to_giocatore = {}

        try:
            f = open("giocatori.txt", "r")
            line = f.readline().strip()

            while line != "":
                codice = int(line)
                tipo = f.readline().strip()
                nome_cognome = f.readline().strip()
                line = f.readline().strip()

                if tipo == "testa-di-serie":
                    tok = line.split()
                    ranking = int(tok[0])
                    eta = int(tok[1])
                    professionista = tok[2]
                    line = f.readline().strip()

                    tds = TestaDiSerie(codice, tipo, nome_cognome, ranking, eta, professionista)
                    giocatori.append(tds)
                    cod_to_giocatore[codice] = tds

                else:
                    circolo = line
                    line = f.readline().strip()
                    tok = line.split()
                    vinti = int(tok[0])
                    disputate = int(tok[1])
                    line = f.readline().strip()

                    norm = Normale(codice, tipo, nome_cognome, circolo, vinti, disputate)
                    giocatori.append(norm)
                    cod_to_giocatore[codice] = norm

                line = f.readline().strip()

        except IOError as i:
            print(i)
        except Exception as e:
            print(e)

        print("nome, codice, tipo, ranking, età, professionista, circolo, n. set vinti, n. partite")

        for g in giocatori:
            print(g)
        print()

        try:
            f = open("partite.txt", "r")
            line = f.readline().strip()

            while line != "":
                tok = line.split()
                codice_primo_giocatore = int(tok[0])
                codice_secondo_giocatore = int(tok[1])
                line = f.readline().strip()
                tok = line.split()
                set_vinti_primo_giocatore = int(tok[0])
                set_vinti_secondo_giocatore = int(tok[1])
                inc_per_primo_giocatore = Incontro(codice_primo_giocatore, codice_secondo_giocatore, set_vinti_primo_giocatore, set_vinti_secondo_giocatore)
                cod_to_giocatore[codice_primo_giocatore].add_incontro(inc_per_primo_giocatore)
                inc_per_secondo_giocatore = Incontro(codice_secondo_giocatore, codice_primo_giocatore, set_vinti_secondo_giocatore, set_vinti_primo_giocatore)
                cod_to_giocatore[codice_secondo_giocatore].add_incontro(inc_per_secondo_giocatore)

                if set_vinti_primo_giocatore > set_vinti_secondo_giocatore:
                    cod_to_giocatore[codice_primo_giocatore].add_partita_vinta()
                else:
                    cod_to_giocatore[codice_secondo_giocatore].add_partita_vinta()

                line = f.readline().strip()

        except IOError as i:
            print(i)
        except Exception as e:
            print(e)

        for gioc in giocatori:
            print(gioc.stampa_incontri())
        print()

        max_vincite = 0
        nome_giocatore_max_vincite = ""

        for gioca in giocatori:
            if gioca.get_partite_vinte() > max_vincite:
                max_vincite = gioca.get_partite_vinte()
                nome_giocatore_max_vincite = gioca.get_nome_giocatore()

        print("Il giocatore che ha vinto più partite è " + nome_giocatore_max_vincite)


if __name__ == "__main__":
    GestioneTorneo.main()
